distribution/lib/Standard/Database/0.0.0-dev/src/DB_Column.enso

```diff
@@ -209,6 +209,106 @@ type DB_Column
+       query returned by the callback inside of this scope. See the examples
+       below to see how the generated SQL is structured.
+
+       Internally, `let` generates a unique name for the CTE, and creates a
+       "reference" column which refers to that unique name. This
+       "reference" column is passed to the callback, which can compute values
+       based on the reference and any other values. Finally, the return value of
+       the callback is wrapped in the binding `WITH ... AS` clause, which is
+       returned from the original call to `let`.
+
+       `let` is only available in database backends that support `WITH` clauses
+       inside expressions. For database backends that only support a single
+       `WITH` claus at the top level, `let` returns `self` unchanged.
+
+       Semantically, the following expressions will always have the same value:
+
+           1. f column
+           2. column.let f
+
+       ? When to use `let`
+
+         `let` can make queries shorter and/or simpler by eliminating
+         duplicates. However, the `WITH` clause itself, including the bound CTE
+         table name, also takes up space, so if the `self` argument to `let`
+         isn't very large, `let` can actually make the query longer.
+
+         For this reason, it is generally better to apply `let` to values
+         before passing them into a library method, rather than to apply it to
+         the value inside the library method itself.
+
+       > Example
+         Remove duplicates of a large column expression from a query.
+
+            ## Without CTEs
+
+             column = table_builder [['x', [1.3, 4.7, -1.3, -4.7]]] . at "x"
+             rounded = column.round
+             large = rounded * rounded
+             large.to_sql
+             ## =>
+
+                -- Two copies of the complex rounding query
+                SELECT ... [complex rounding query] * [complex rounding query]
+                    ... FROM ...
+
+            ## With CTEs
+
+             not_so_large = column.round.let rounded->
+                rounded * rounded
+             not_so_large.to_sql
+             ## =>
+
+                -- One copy of the complex rounding query
+                SELECT ... (WITH temp_table as ([complex rounding query])
+                                 temp_table.x * temp_table.x)
+                    ... FROM ...
+
+       > Example
+         Use multiple CTEs in a query.
+
+         (column_a * column_b).let product_a_b->
+            (product_a_b * 10).let times_ten->
+                times_ten + product_a_b + 100
+
+       > Example
+         Give names to the CTE table names.
+
+         (column_a * column_b).let name="product_a_b" product_a_b->
+            (product_a_b * 10).let name="times_ten" times_ten->
+                times_ten + product_a_b + 100
+    let : (DB_Column -> DB_Column) -> Text -> DB_Column
+    let self (callback : (DB_Column -> DB_Column)) (name : Text = "let") -> DB_Column =
```

Comment:
I think I'd reverse the order of arguments and make the name required. It will make the queries more readable if helpful names are always provided.

Reply:
I think we are missing `Arguments:` section?

What if we rename `name` to `name_hint` and use it as suggested prefix in `generate_random_table_name`? Then the generated names would contain it, making it easier to discern these names.

Reply:
I switched the arguments and added Arguments:.

I originally used the name to generate the table name, but since the table name is replaced in Base_Generator, I am not only using the name there.